tokio/src/time/throttle.rs

```diff
@@ -16,17 +16,27 @@ use std::{
 #[derive(Debug)]
 #[must_use = "streams do nothing unless polled"]
 pub struct Throttle<T> {
-    delay: Delay,
+    /// `None` when duration is zero.
+    delay: Option<Delay>,
```

Comment:
I found a "bug" that was hidden by the fact that "mock clock" tests always used ms rounded numbers for time values. When I switched to the `time::pause()` style, the time was paused at sub ms granularity. In this case, when throttling with duration zero, the stream was still throttled at up to a ms granularity.

The simplest fix was to switch the delay to `Option` and use `None` when throttling at duration zero.